forge/src/main/java/org/valkyrienskies/mod/forge/mixin/compat/sodium/MixinSodiumShipRenderer.java

```diff
@@ -0,0 +1,101 @@
+package org.valkyrienskies.mod.forge.mixin.compat.sodium;
+
+import org.joml.Matrix4d;
+import org.joml.Matrix4dc;
+import org.joml.Matrix4f;
+import org.joml.Vector3d;
+import org.joml.Vector3dc;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.valkyrienskies.mod.mixinducks.mod_compat.sodium.RenderSectionManagerDuck;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.PoseStack;
+
+import me.jellysquid.mods.sodium.client.gl.device.CommandList;
+import me.jellysquid.mods.sodium.client.gl.device.RenderDevice;
+import me.jellysquid.mods.sodium.client.render.SodiumWorldRenderer;
+import me.jellysquid.mods.sodium.client.render.chunk.ChunkRenderMatrices;
+import me.jellysquid.mods.sodium.client.render.chunk.RenderSectionManager;
+import me.jellysquid.mods.sodium.client.render.chunk.terrain.DefaultTerrainRenderPasses;
+import me.jellysquid.mods.sodium.client.render.chunk.terrain.TerrainRenderPass;
+import me.jellysquid.mods.sodium.client.render.viewport.CameraTransform;
+
+import org.valkyrienskies.core.api.ships.properties.ShipTransform;
+import org.valkyrienskies.mod.common.VSClientGameUtils;
+import org.valkyrienskies.mod.common.hooks.VSGameEvents;
+import org.valkyrienskies.mod.common.hooks.VSGameEvents.ShipRenderEventSodium;
+import org.valkyrienskies.mod.mixin.mod_compat.sodium.RenderSectionManagerAccessor;
+
+import net.minecraft.client.renderer.RenderType;
+
+@Mixin(value = SodiumWorldRenderer.class, remap = false, priority = 1100)
+public class MixinSodiumShipRenderer {
```

Comment:
Please name this MixinSodiumWorldRenderer to follow naming convention

Reply:
i could in theory just put the vsRenderLayer stuff in common but idk if that should be public or not?